### PR TITLE
Web ort build all script

### PR DESCRIPTION
**Pull request by shalvamist**
**Description**: Describe your changes.
Adding a python script to simplify the initial build of ORT Web - currently supports - 
1. Building the WASM artifacts
2. Automatic copy of the artifacts to destination folders
3. NPM build

to run the script you will need this command - python .\js\buildAll.py -b Release -c Release
note that you still need to install the prerequisites
 
**Motivation and Context**
Making the first build of ORT Web a smoother process

### CODE REVIEW

**Review by fs-eire — CHANGES_REQUESTED** [review]
1. Please revise the scripts to resolve the pyflakes and linter warnings

2. comments about build flags:
    - no need to use '--emsdk_version'. The reason why CI need to use it is because the CI uses an environment with specific Java/Nodejs version. for normal users, we don't specify the emsdk version. let it use the default emsdk version.
    - it probably needs a discussion about `--cmake_generator "Visual Studio 16 2019"'`. by default CMake should be able to locate the correct installed vs. using this flag may cause those who doesn't install the exact same version fail the build. and this is not working in UNIX
    - try to align with the build.py. for example, use '--config' instead of '-b' for build configurations (Debug/Release/RelWithDebInfo)
    - use a '-h'/'--help'. (together with parser.add_argument() blocks and highlight them) this does not only help the users but also help people who read and maintain this script

4. comments about compatibility:
    - support UNIX
        - use [os.pathsep](https://docs.python.org/3/library/os.html#os.pathsep)
    - support spaces in path. always wrap parameters with quotes (`"`)
    - always use all lowercase in path to avoid compatibility issue ( windows is case-insensitive and UNIX is case-sensitive )
    - call build.sh instead of build.bat in UNIX
    - don't use os.system(). always use child_process
6. general recommendations:
    - always call '--skip_submodule_sync'. because the shell script needs to prepare emsdk first.
    - pass all rest commandline parameters to build.py

File: js/buildAll.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+parser = argparse.ArgumentParser()
+parser.add_argument("-b", "--build", choices=["Release", "Debug"], help="build WASM - {Release,Debug}")
+parser.add_argument("-c", "--copy", choices=["Release", "Debug"], help="copy WASM artifacts - {Release,Debug}")
+args = parser.parse_args()
+configuration = "none"
+
+if args.build:
+    configuration = args.build
+    command = (
+        "build.bat " + firsttime_flags + " --config " + configuration + " --build_dir " + BinariesDirectory + "\\wasm"
```
Comment (fs-eire) [inline js/buildAll.py:48]: i think there is no need to use different lists for firsttime_flags and default_flags. the only difference is --path_to_proto_c, you can append this in the for loop. (--skip_submodule_sync should be the same, as discussed in the other comment)

File: js/buildAll.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            + build_flags[i]
+        )
+        # print(command)
+        os.system(command)
+
+# copy the files to the right location
+if args.copy:
+    configuration = "\\" + args.copy + "\\"
+    ## Copying WASM artifacts
+    file_name = "ort-wasm.wasm"
```
Comment (fs-eire) [inline js/buildAll.py:70]: how about using a list of filenames and copy them in a loop?

Comment (shalvamist) [reply js/buildAll.py]: yep - I agree. I updated the script to use a loop

File: js/buildAll.py (hunk 1 above)
Comment (shalvamist) [reply js/buildAll.py]: Thanks for catching this - I updated the default flags and now we have only one.

File: js/buildAll.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+from os.path import abspath, dirname
+
+build_script = "build.sh"
+if "win" in sys.platform:
```
Comment (fs-eire) [inline js/buildAll.py:9]: use `sys.platform.startswith("win")`

do you know that macOS platform name is "darwin" ?

File: js/buildAll.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    build_script = "build.bat"
+
+build_dir = ["wasm", "wasm_SIMD", "wasm_threaded", "wasm_SIMD_threaded"]
+build_flags = ["", " --enable_wasm_simd", " --enable_wasm_threads", " --enable_wasm_simd --enable_wasm_threads "]
```
Comment (fs-eire) [inline js/buildAll.py:13]: please remove the space inside the flags. when you join those flags always add spaces

File: js/buildAll.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+wasm_file_names = ["ort-wasm.wasm", "ort-wasm-simd.wasm", "ort-wasm-threaded.wasm", "ort-wasm-simd-threaded.wasm"]
+js_file_names = ["ort-wasm.js", "ort-wasm-threaded.worker.js", "ort-wasm-threaded.js"]
+build_dir_js = ["wasm", "wasm_threaded", "wasm_threaded"]
+
+default_flags = " --build_wasm --skip_tests --skip_submodule_sync --parallel "
+BinariesDirectory = ".\\build\\"
```
Comment (fs-eire) [inline js/buildAll.py:19]: use consistent naming convention for consts. you have multiple in this file:

default_flags
BinariesDirectory
Dist_Path

File: js/buildAll.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+Dist_Path = ".\\js\\web\\dist\\"
+Binding_Path = ".\\js\\web\\lib\\wasm\\binding\\"
+
+# Stage #1 - Creating the folders for the build process
+if not os.path.isdir(BinariesDirectory):
+    os.mkdir(BinariesDirectory)
+
+for path in build_dir:
+    dirctory_name = BinariesDirectory + path
```
Comment (fs-eire) [inline js/buildAll.py:28]: use join

**Comment by shalvamist** [comment]
Closing this PR - moving to a new PR - https://github.com/microsoft/onnxruntime/pull/12643
Introduced a few more checks to the python script and change the folder structure a bit to align with the latest code base

**Pull request closed without merging**